src/ring_theory/integral_closure.lean

```diff
@@ -0,0 +1,280 @@
+/-
+Copyright (c) 2019 Kenny Lau. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kenny Lau
+
+Integral closure of a subring.
+-/
+
+import ring_theory.adjoin
+
+universes u v
+
+open polynomial submodule
+
+variables (R : Type u) {A : Type v}
+variables [comm_ring R] [comm_ring A]
+variables [decidable_eq R] [decidable_eq A]
+variables [algebra R A]
+
+def is_integral (x : A) : Prop :=
+∃ p : polynomial R, monic p ∧ aeval R A x p = 0
+
+variables {R}
+theorem is_integral_algebra_map {x : R} : is_integral R (algebra_map A x) :=
+⟨X - C x, monic_X_sub_C _,
+by rw [alg_hom.map_sub, aeval_def, aeval_def, eval₂_X, eval₂_C, sub_self]⟩
+
+theorem is_integral_of_subring {x : A} (T : set R) [is_subring T]
```

Comment:
The name `is_integral_of_is_integral_over_subring` is quite long, but a lot more descriptive...